product_logistics_uom/models/product_template.py

```diff
@@ -0,0 +1,161 @@
+        "uom.uom",
+        string="Volume Unit of Measure",
+        domain=lambda self: [
+            ("category_id", "=", self.env.ref("uom.product_uom_categ_vol").id)
+        ],
+        default=lambda self: self._get_volume_uom_id_from_ir_config_parameter(),
+    )
+    volume_uom_name = fields.Char(
+        string="Volume unit of measure label",
+        related="volume_uom_id.name",
+        readonly=True,
+    )
+
+    weight_uom_id = fields.Many2one(
+        "uom.uom",
+        string="Weight Unit of Measure",
+        domain=lambda self: [
+            ("category_id", "=", self.env.ref("uom.product_uom_categ_kgm").id)
+        ],
+        compute=False,
+        default=lambda self: self._get_weight_uom_id_from_ir_config_parameter(),
+    )
+
+    weight_uom_name = fields.Char(
+        string="Weight unit of measure label",
+        related="weight_uom_id.name",
+        readonly=True,
+    )
+
+    show_volume_uom_warning = fields.Boolean(
+        help="Technical field used to warn the user to change the volume"
+        "uom since the value for product_volume is too small and has been"
+        "rounded.",
+        compute="_compute_show_volume_uom_warning",
+    )
+
+    show_weight_uom_warning = fields.Boolean(
+        help="Technical field used to warn the user to change the weight"
+        "uom since the value for product_weight is too small and has been"
+        "rounded.",
+        compute="_compute_show_weight_uom_warning",
+    )
+
+    @api.model
+    def _get_volume_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_volume_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_volume_uom_id_from_ir_config_parameter()
+
+    @api.model
+    def _get_weight_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_weight_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_weight_uom_id_from_ir_config_parameter()
+
+    @api.model
+    def _get_length_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_length_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_length_uom_id_from_ir_config_parameter()
+
+    @api.depends(
+        "product_variant_ids",
+        "product_variant_ids.product_volume",
+        "volume",
+        "volume_uom_id",
+    )
+    def _compute_product_volume(self):
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
```

Comment:
```suggestion
            lambda template: template.product_variant_count == 1
```
better with using product_variant_count was caculated in https://github.com/odoo/odoo/blob/69b404c7109ff689381f56520aad758424ec01aa/addons/product/models/product_template.py#L399